src/Symfony/Component/Filesystem/Tests/PathTest.php

```diff
@@ -0,0 +1,1341 @@
+    {
+        Path::makeAbsolute('css/style.css', array());
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The base path "webmozart/puli" is not an absolute path.
+     */
+    public function testMakeAbsoluteFailsIfBasePathNotAbsolute()
+    {
+        Path::makeAbsolute('css/style.css', 'webmozart/puli');
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The base path must be a non-empty string. Got: ""
+     */
+    public function testMakeAbsoluteFailsIfBasePathEmpty()
+    {
+        Path::makeAbsolute('css/style.css', '');
+    }
+
+    /**
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessage The base path must be a non-empty string. Got: NULL
+     */
+    public function testMakeAbsoluteFailsIfBasePathNull()
+    {
+        Path::makeAbsolute('css/style.css', null);
+    }
+
+    public function provideAbsolutePathsWithDifferentRoots()
+    {
+        return array(
+            array('C:/css/style.css', '/webmozart/puli'),
+            array('C:/css/style.css', '\\webmozart\\puli'),
+            array('C:\\css\\style.css', '/webmozart/puli'),
+            array('C:\\css\\style.css', '\\webmozart\\puli'),
+
+            array('/css/style.css', 'C:/webmozart/puli'),
+            array('/css/style.css', 'C:\\webmozart\\puli'),
+            array('\\css\\style.css', 'C:/webmozart/puli'),
+            array('\\css\\style.css', 'C:\\webmozart\\puli'),
+
+            array('D:/css/style.css', 'C:/webmozart/puli'),
+            array('D:/css/style.css', 'C:\\webmozart\\puli'),
+            array('D:\\css\\style.css', 'C:/webmozart/puli'),
+            array('D:\\css\\style.css', 'C:\\webmozart\\puli'),
+
+            array('phar:///css/style.css', '/webmozart/puli'),
+            array('/css/style.css', 'phar:///webmozart/puli'),
+
+            array('phar://C:/css/style.css', 'C:/webmozart/puli'),
+            array('phar://C:/css/style.css', 'C:\\webmozart\\puli'),
+            array('phar://C:\\css\\style.css', 'C:/webmozart/puli'),
+            array('phar://C:\\css\\style.css', 'C:\\webmozart\\puli'),
+        );
+    }
+
+    /**
+     * @dataProvider provideAbsolutePathsWithDifferentRoots
+     */
+    public function testMakeAbsoluteDoesNotFailIfDifferentRoot($basePath, $absolutePath)
+    {
+        // If a path in partition D: is passed, but $basePath is in partition
+        // C:, the path should be returned unchanged
+        $this->assertSame(Path::canonicalize($absolutePath), Path::makeAbsolute($absolutePath, $basePath));
+    }
+
+    public function provideMakeRelativeTests()
+    {
+        $paths = array_map(function (array $arguments) {
+            return array($arguments[2], $arguments[1], $arguments[0]);
+        }, $this->providePathTests());
+
+        return array_merge($paths, array(
+            array('/webmozart/puli/./css/style.css', '/webmozart/puli', 'css/style.css'),
```

Comment:
Not sure how Bernhard feels with this change, but maybe we should use something like symfony here instead of webmozart and/or puli. Or some paths like in the tests below...

Reply:
I replaced `puli` with `symfony`. I think keeping `webmozart` is fine